patches/minecraft/net/minecraft/block/BlockRedstoneTorch.java.patch

```diff
@@ -0,0 +1,56 @@
+--- ../src-base/minecraft/net/minecraft/block/BlockRedstoneTorch.java
++++ ../src-work/minecraft/net/minecraft/block/BlockRedstoneTorch.java
+@@ -2,9 +2,11 @@
+ 
+ import com.google.common.collect.Lists;
+ import com.google.common.collect.Maps;
++
+ import java.util.List;
+ import java.util.Map;
+ import java.util.Random;
++
+ import net.minecraft.block.state.IBlockState;
+ import net.minecraft.creativetab.CreativeTabs;
+ import net.minecraft.init.Blocks;
+@@ -121,11 +123,24 @@
+         {
+             list.remove(0);
+         }
++        
++        int oldCurrent = this.field_150113_a ? 15 : 0;
```

Comment:
This function is called on both the client and server, so there is still the desync issue.